src/components/ListingDetails/index.tsx

```diff
@@ -0,0 +1,320 @@
+      return d.adjudicating_agency ? parseAgency(d.adjudicating_agency) : 'N/A';
+      // (data as CaseListing).adjudicating_agency ? parseAgency(data.adjudicating_agency) : 'N/A',
+    },
+  },
+  // Client Languages
+  {
+    label: 'Client Language(s)',
+    getValue: (data: Listing) => (data as CaseListing).languages.join(', '),
+  },
+  // Client Country of Origin
+  {
+    label: 'Client Country of Origin',
+    getValue: (data: Listing) => (data as CaseListing).country || 'N/A',
+  },
+  // Client Location
+  {
+    label: 'Client Location',
+    getValue: (data: Listing) => (data as CaseListing).client_location || 'N/A',
+  },
+  // Custody Location
+  {
+    // Temporary
+    label: 'Custody Location',
+    getValue: () => 'San Diego, CA',
+  },
+];
+
+const caseInterpretationFields = [
+  // Time Commitment
+  {
+    label: 'Time Commitment',
+    getValue: (data: Listing) =>
+      parseTimeCommitment(
+        (data as CaseListing).hours_per_week,
+        (data as CaseListing).num_weeks,
+      ),
+  },
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: (data: Listing) => (data as CaseListing).languages.join(', '),
+  },
+  // Remote/In Person
+  {
+    label: 'Remote/In Person',
+    getValue: (data: Listing) =>
+      (data as CaseListing).is_remote ? 'Remote' : 'In Person',
+  },
+];
+
+const intepretationFields = [
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: (data: Listing) => (data as Interpretation).languages.join(', '),
+  },
+  // Interpretation Type
+  {
+    label: 'Interpretation Type', // 'Language Support Type' for consistency w/ Doc?
+    getValue: (data: Listing) => 'One-Time Intepretation',
+  },
+  // Remote/In Person
+  {
+    label: 'Remote/In Person',
+    getValue: (data: Listing) =>
+      (data as Interpretation).is_remote ? 'Remote' : 'In Person',
+  },
+];
+
+const docFields = [
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: (data: Listing) =>
+      (data as DocumentTranslation).languages.join(', '),
+  },
+  // Language Support Type
+  {
+    label: 'Language Support Type',
+    getValue: (data: Listing) => 'Written',
+  },
+  // Number of Pages
+  {
+    label: 'Number of Pages',
+    getValue: (data: Listing) => (data as DocumentTranslation).num_pages,
+  },
+];
+
+const LCAFields = [
+  // Country Field
+  {
+    label: 'Country Field',
+    getValue: (data: Listing) => (data as LimitedCaseAssignment).country,
+  },
+  // Language(s)
+  {
+    label: 'Language(s)',
+    getValue: (data: Listing) =>
+      (data as LimitedCaseAssignment).languages.join(', '),
+  },
+  // Expected Deliverable
+  {
+    label: 'Expected Deliverable',
+    getValue: (data: Listing) => (data as LimitedCaseAssignment).deliverable,
+  },
+];
+
+export default function ListingDetails({
+  listingData,
+}: {
+  listingData: Listing;
+}) {
+  const listingFields = () => {
+    if (listingData.listing_type === 'CASE') {
+      if (listingData.needs_interpreter) {
```

Comment:
I think we may want a different if-check for case interpretations. Based off of this implementation, we'd always be returning `caseInterpretationFields` when `listingData.needs_interpreter == true` even if we intend to return `caseFields`. We should instead use an optional `interpretation` prop to help indicate that we're dealing with case interpretations. The `ListingCard` component uses this approach, so it might be helpful to take a quick look at it for reference!

Reply:
I think this should also answer the question you left in the comment below